src/main/java/org/jboss/ejb/protocol/remote/RemoteEJBReceiver.java

```diff
@@ -112,10 +110,8 @@ public void handleFailed(final IOException exception, final EJBReceiverInvocatio
                 Logs.REMOTING.error("Error in connecting to " + destination + " : Please check if the client and server are configured to use the same protocol and ports.");
             } else if (exception instanceof SSLException && exception.getMessage().equals("Unrecognized SSL message, plaintext connection?")) {
                 Logs.REMOTING.error("Error in connecting to " + destination + " : The destination doesn't support SSL. Did you mean to use http protocol instead?");
-            } else if (exception instanceof SSLHandshakeException && exception instanceof ConnectException ) {
-                Logs.REMOTING.error("Error while connecting @ " + destination + " becauese of incorrect protocol and port combination compare to the server.");
             }
-            attachment.requestFailed(new RequestSendFailedException("Destination @ " + destination, exception, false), retryExecutorWrapper.getExecutor(Endpoint.getCurrent().getXnioWorker()));
+            attachment.requestFailed(new RequestSendFailedException("Check the Destination @" + destination +" protocol and port combination compare to the server ", exception, false), retryExecutorWrapper.getExecutor(Endpoint.getCurrent().getXnioWorker()));
```

Comment:
I think the message is from the first `Logs.REMOTING.error` is a bit better and consistent:
```java
"Error in connecting to " + destination + " : Please check if the client and server are configured to use the same protocol and ports."
```

Reply:
I have tested a few times but looks like `Logs.REMOTING.error` is taking care of HTTP Upgrade cases, not for all port+protocol mismatch scenarios.

Reply:
your new message: "protocol and port combination compare to the server" means the same as the first one (as I pasted above). And the first one appears more readable to me.

my point is to keep the last line of your change (`attachment.requestFailed(...)`), using the message from the first line.